src/Cake.Core/CakeEnvironment.cs

```diff
@@ -53,6 +53,27 @@ public bool IsUnix()
             return Machine.IsUnix();
         }
 
+        private static readonly string[] _debugArgs = 
+        {
+            "-d", "--d", "-d=true", "--d=true",
+            "-debug", "--debug", "-debug=true", "--debug=true"
+        };
+
+        /// <summary>
+        /// Gets a value indicating whether the current execution is running in debug.
+        /// </summary>
+        /// <returns>
+        /// Whether or not the current execution is running in debug.
+        /// </returns>
+        public bool IsDebug
+        {
+            get
+            {
+                var args = Environment.GetCommandLineArgs();
+                return args.Intersect(_debugArgs).Any();
+            }
+        }
```

Comment:
@patriksvensson yes, I could add this as an separate PR as it solves another issue..